tamr_toolbox/workflow/jobs.py

```diff
@@ -188,3 +189,29 @@ def get_upstream_projects(project: Project) -> List[Project]:
     upstream_projects = _find_upstream_projects(project)
 
     return upstream_projects
+
+
+def get_enrichment_project_output_dataset(project: Project) -> str:
+    """Check for the output dataset for a specified enrichment project
+
+    Args:
+        project: the tamr enrichment project for which associated output dataset name is retrieved
+    """
+    if project.type == "ENRICHMENT":
+        input_dataset_id = [d for d in project.input_datasets()][0].relative_id.split("/")[-1]
+        usage_json = project.client.get(
+            endpoint=f"/api/versioned/v1/datasets/{input_dataset_id}/usage",
+            headers={"Content-Type": "application/json", "Accept": "application/json"},
+        )
+        usage_json.raise_for_status()
+        output_dataset_name = [
+            dep["datasetName"]
+            for dep in json.loads(usage_json.text)["dependencies"]
+            if "outputFromProjectSteps" in dep
+            and dep["outputFromProjectSteps"]
+            and "projectName" in dep["outputFromProjectSteps"][0]
+            and dep["outputFromProjectSteps"][0]["projectName"] == project.name
+        ][0]
```

Comment:
Hi @Kovacsova - I developed this func on a customer (WCG) instance where the input to the enrichment project was the UD of another project. In these cases, when we get the dependencies of the enrichment project `input` dataset, there will be some upstream dependencies that would not have the `outputFromProjectSteps` tag. In the case of the enrichment workflow that is set up in the TBOX Test instance though, the input to the enrichment project is a source dataset from `csv` so its dependency will be a *downstream* dependency and I agree that the above-mentioned checks would become trivial for this special case.

Reply:
Overall, this could be more concise with the dictionary `get` method:
```
output_dataset_names = [
    dep["datasetName"]
    for dep in json.loads(usage_json.text)["dependencies"]
    if dep.get("outputFromProjectSteps", [{}]).get("projectName") == project.name
]
if len(output_dataset_names) == 1:
    output_dataset_name = output_dataset_names[0]
else:
    raise SomeException("...")
```

Reply:
Wow - that above alternative looks very neat!

Reply:
@skalish : just so I am sure there are no lose ends here, if `outputFromProjectSteps` is not in `dep` then 
`dep.get("outputFromProjectSteps", [{}]) we would return `[{}]`, and now if we are to call `get("projectName")` on the this list of empty dict object, would that be okay? Shouldn't we pass in a `{}` instead of `[{}]`?

Reply:
Could this work?

```
output_dataset_name = [dep["datasetName"] 
for dep in json.loads(usage_json.text)["dependencies"]
if ("test1" in [elem.get("projectName") for elem in dep.get("outputFromProjectSteps", [{}])])]

if len(output_dataset_names) == 1:
    output_dataset_name = output_dataset_names[0]
else:
    raise SomeException("...")
```

Reply:
Hi @Kovacsova - Sam and I met over this ticket yesterday and decided to expand the scope to include all project types. I will have to resubmit this PR.